### PR TITLE
add keycode for all events

### PR DESCRIPTION
**Pull request by meeuw**
Same trick as for `KeyEvent`, translate code to keycode in constructor.

Also added value to Rel/AbsEvent.

### CODE REVIEW

**Comment by sezanzeb** [comment]
I see, thanks. I'd be in favor since it adds consistency.

I wonder why `keycode` was added to `KeyEvent` in the first place though, because it seems redundant and apparently only exists for human readability. But removing the public `keycode` member might break peoples code at this point.

---

Is it possible to avoid this duplicate code for all `*Event` classes?

```py
        try:
            self.keycode = REL[event.code]
        except KeyError:
            if allow_unknown:
                self.keycode = '0x{:02X}'.format(event.code)
            else:
                raise
```

File: evdev/events.py [hunk 1]
```diff
@@ -169,15 +185,23 @@ class SynEvent(object):
     the multitouch protocol.
     '''
 
-    __slots__ = 'event'
+    __slots__ = 'event', 'keycode'
+
+    def __init__(self, event, allow_known):
```
Comment (gvalkov) [inline evdev/events.py:190]: Should be `allow_unknown`, no?

**Review by gvalkov — CHANGES_REQUESTED** [review]

File: evdev/events.py (hunk 1 above)
Comment (meeuw) [reply evdev/events.py]: whups, yes, fixed in f24f7f48843876c149ebda3f18842ec5db21d5d2

**Comment by meeuw** [comment]
I'm not so sure about the `allow_unknown` altogether.

Might be a good idea to refactor like this:

```python
if not allow_unkown:
   if not event.code in keys: raise KeyError
self.keycode = keys.get(event.code, '0x{:02X}'.format(event.code))
```

**Comment by gvalkov** [comment]
Hm, I'm honestly not sure which is more expensive: the `try/except` or looking up the value in `keys` twice :)

Also, `keycode` as a term is only relevant for `KeyEvent`s (i.e. scancode -> keycode). It should probably be `event_code` or something like that.

**Comment by gvalkov** [comment]
Ok, so I had this merged, but I think I'm going to revert it. All in all, I'm just not sure if it's necessary. It adds extra processing in possibly tight loops and it's mostly just useful for the `__str__()` of the event.

Thanks for effort nonetheless!

**Comment by meeuw** [comment]
Thanks for considering it. I think I'll avoid the keycode altogether and use `evdev.ecodes.keys[event.event.code]` instead.

**Pull request closed without merging**